types/src/v2/params.rs

```diff
@@ -335,4 +327,24 @@ mod test {
 			assert_eq!(&serialized, initial_ser);
 		}
 	}
+
+	#[test]
+	fn subscription_params_serialize_work() {
+		let ser =
+			serde_json::to_string(&JsonRpcSubscriptionParams { subscription: SubscriptionId::Num(12), result: "goal" })
+				.unwrap();
+		assert_eq!(ser, r#"{"subscription":12,"result":"goal"}"#);
+	}
+
+	#[test]
+	fn subscription_params_deserialize_work() {
+		let ser = r#"{"subscription":"9","result":"offside"}"#;
+		assert!(
+			serde_json::from_str::<JsonRpcSubscriptionParams<()>>(ser).is_err(),
+			"invalid type should not be deser"
```

Comment:
Will the average user know that `deser` is `deserialize`?

Reply:
changed it, probably better to verbose in this case thanks.

Reply:
Always good to be explicit, but yeah, most readers would be familiar with the abbreviation.